modules/streams/src/main/scala/dev/profunktor/redis4cats/interpreter/pubsub/Subscriber.scala

```diff
@@ -43,7 +43,8 @@ class Subscriber[F[_]: ConcurrentEffect: ContextShift: Log, K, V](
 
   override def unsubscribe(channel: RedisChannel[K]): Stream[F, Unit] =
     Stream.eval {
-      JRFuture(Sync[F].delay(subConnection.async().unsubscribe(channel.underlying))).void
+      state.update(_ - channel.underlying) *>
+        JRFuture(Sync[F].delay(subConnection.async().unsubscribe(channel.underlying))).void
```

Comment:
We could make sure to remove the topic once we have unsubscribed. Something like:

```scala
JRFuture(... unsubscribe ...).guarantee(state.update(_ - channel.underlying))
```

Reply:
I think we would need to send a `none[V]` message as well to end the subscriber stream (it uses `unNone`):

```scala
topic.publish1(none[V])
```

Not sure if we have access to it here, though (we should).

Reply:
I think we should use unNoneTerminate on subscriber stream, instead of continuing using unNone

Reply:
Fixed

Reply:
We need `guarantee` there not `*>`.

Reply:
Sorry, I didn't know that there is such method. Fixed again :)

Reply:
No problems, thanks!